src/responder/gasQueue.ts

```diff
@@ -286,14 +287,55 @@ export class GasQueue {
         return new GasQueue(clonedArray, this.emptyNonce, this.replacementRate, this.maxQueueDepth);
     }
 
-    //TODO:198: documentation and tests
-    public unlock(requests: GasQueueItemRequest[]): GasQueue {
-        // we need to keep track of the max gas price used at a given nonce
-        // since we need to go at least higher than this
+    /**
+     * Returns all queue items that are in this queue but not in the supplied queue
+     * @param otherQueue
+     */
+    public difference(otherQueue: GasQueue): GasQueueItem[] {
+        return this.queueItems.filter(tx => !otherQueue.queueItems.includes(tx));
+    }
+
+    /**
+     * Checks to see if this queue contains an item with the supplied identifier
+     * @param queueItem
+     */
+    public contains(identifier: PisaTransactionIdentifier): boolean {
+        return this.queueItems.findIndex(i => i.request.identifier.equals(identifier)) !== -1;
+    }
+
+    
+    /**
+     * Re-add some items that have lower nonces than any in the current queue
+     * @param lowerNonceItems 
+     */
+    public unlock(lowerNonceItems: GasQueueItem[]): GasQueue {
```

Comment:
The name `unlock` is rather confusing, as it doesn't match what the code does and it's not clear in what sense items are "locked".

Reply:
Changed to prepend